devchat/_cli/utils.py

```diff
@@ -111,30 +110,37 @@ def clone_git_repo(target_dir: str, repo_urls: List[str]):
     raise GitCommandError(f"Failed to clone repository to {target_dir}")
 
 
-def convert_yml(config_json_file, config_yml_file):
-    with open(config_json_file, 'r', encoding='utf-8') as ofile:
-        config_value = json.load(ofile)
-        new_config_value = {"default_model": "gpt-3.5-turbo"}
-        new_config_value["models"] = {}
-        new_config_value["models"][config_value["model"]] = {
-            "stream": config_value["OpenAI"]["stream"],
-            "temperature": config_value["OpenAI"]["temperature"]
-		}
+def parse_legacy_config(config_json_file) -> Tuple[str, OpenAIModelConfig]:
+    with open(config_json_file, 'r', encoding='utf-8') as file:
+        legacy_data = json.load(file)
+        if 'model' not in legacy_data:
+            return None, None
 
-        with open(config_yml_file, 'w', encoding='utf-8') as file:
-            yaml.dump(new_config_value, file)
+        model = legacy_data['model']
+        config = OpenAIModelConfig()
+        if 'tokens-per-prompt' in legacy_data:
+            config.max_input_tokens = legacy_data['tokens-per-prompt']
+        if 'OpenAI' in legacy_data:
+            if 'temperature' in legacy_data['OpenAI']:
+                config.temperature = legacy_data['OpenAI']['temperature']
+            if 'stream' in legacy_data['OpenAI']:
+                config.stream = legacy_data['OpenAI']['stream']
+        return model, config
 
 
 def get_model_config(repo_chat_dir: str, user_chat_dir: str,
-                     model: Optional[str] = None) -> Tuple[str, ModelConfig]:
+                     model: Optional[str] = None) -> Tuple[str, Any]:
+    manager = ConfigManager(user_chat_dir)
```

Comment:
@bobo Here we deal with creation of config.yml with default settings if it doesn't exist.